protocols/perf/src/bin/perf-client.rs

```diff
@@ -43,15 +44,217 @@ async fn main() -> Result<()> {
 
     let opts = Opts::parse();
 
-    info!(
-        "{}",
-        format!(
-            "Initiating performance tests with {}\n",
-            opts.server_address
-        )
-        .bold()
+    let benchmarks: [Box<dyn Benchmark>; 3] = [
+        Box::new(Throughput {}),
+        Box::new(RequestsPerSecond {}),
+        Box::new(ConnectionsPerSecond {}),
+    ];
+
+    for benchmark in benchmarks {
+        info!(
+            "{}",
+            format!("Start benchmark: {}", benchmark.name()).underline(),
+        );
+
+        benchmark.run(opts.server_address.clone()).await?;
+    }
```

Comment:
I find this indirection confusing. Why not just have 3 free-functions and call them in a row? That would remove the need for a trait, a loop and the declaration of the array. Seems a lot simpler to me.

```suggestion
benchmark_throughput(opts.server_address.clone()).await?;
benchmark_requests_per_second(opts.server_address.clone()).await?;
benchmark_connections_per_second(opts.server_address).await?;
```

Reply:
`Benchmark` indirection is removed with [fa6886c](https://github.com/libp2p/rust-libp2p/pull/3646/commits/fa6886cd73d09b9b0ff0b7ce6334a42d760ddf88).